### DISCUSSION TITLE
Can a pdf format quarto document have typst snippets?

### DISCUSSION THREAD

**Discussion by psads-git** [quarto-dev/quarto-cli#9406] [Q&A] [answered]
### Description

This is a speculation of me: Is it possible to have `typst` snippets inside a pdf format `quarto` document?

I suspect not, as I cannot compile the following `quarto` document:

````
---
format: pdf
---

```{=typst}
$
integral 2 x diff x
$
```
````

**Comment by mcanouil** [comment]
No.

`format: pdf` means LaTeX.
Use `format: typst` for Typst(/PDF), see <https://quarto.org/docs/output-formats/typst.html>.

**Reply by psads-git** [reply]
Thanks, @mcanouil, for your answer.

But maybe by calling `pandoc` from inside a `quarto` document (for instance, using `python`), we can convert `typst` code to `latex`.

**Reply by mcanouil** [reply]
Sorry, but why?
Quarto input is markdown and raw blocks matching the chosen format.
By the way, Typst/LaTeX/HTML share the same (or almost) math system, there is no need for raw blocks. If that is your use case, you are only making things complicated for no reason.

If you want to emit OpenXML and translate it as Typst, or Typst to LaTeX, or any conversion like that, then Quarto is not the tool you need.

Also Pandoc reads everything it knows to AST and then writes as it thinks is the best.

**Comment by cscheid** [comment, marked as answer]
Quarto doesn't do that by itself, but you could write a Lua filter that converts one to the other. No promises on this, but something like

````markdown
---
format: pdf
filters:
  - typst_to_latex.lua
---

```{=typst}
$
integral 2 x diff x
$
```
````

```lua
-- typst_to_latex.lua
function RawBlock(raw)
  if raw.format == "typst" then
    return pandoc.RawBlock("latex", pandoc.write(pandoc.read(raw.text, "typst"), "latex"))
  end
end
```

**Reply by psads-git** [reply]
That seems to work fine, @cscheid! Thanks! That is very great!

**Comment by psads-git** [comment]
> Sorry, but why? 

Some reasons:

**Compatibility:** Many researchers, authors, and technical writers already have existing LaTeX or Typst code snippets from previous work or collaborations. Enabling Quarto to read these snippets ensures compatibility with existing workflows and documents, reducing the need for extensive reformatting or rewriting.

**Code Reuse:** If authors or collaborators have already developed LaTeX or Typst code for specific purposes (e.g., custom macros, theorem environments, or specialized diagrams), being able to include these snippets in Quarto documents facilitates code reuse and consistency across different writing projects.

**Interoperability:** Quarto's ability to read LaTeX and Typst snippets can promote interoperability with other tools and software that generate output in these formats. This integration can streamline workflows and enhance collaboration among teams working with different tools.

Maybe that is why it has been created LaTeX support for Typst via package [mitex](https://typst.app/universe/package/mitex).
